common/src/main/java/org/apache/uniffle/common/netty/buffer/FileSegmentManagedBuffer.java

```diff
@@ -61,7 +63,8 @@ public ByteBuffer nioByteBuffer() {
     FileChannel channel = null;
     try {
       channel = new RandomAccessFile(file, "r").getChannel();
-      ByteBuffer buf = ByteBuffer.allocate(length);
+      ByteBuffer buf =
+          preferDirect ? ByteBuffer.allocateDirect(length) : ByteBuffer.allocate(length);
```

Comment:
Do we need to use the same memory allocator with netty server?

Reply:
Here we need a `ByteBuffer` instead of `ByteBuf`, so I think we don't need to use the same allocator with Netty server.